assets/js/base/components/price-slider/index.js

```diff
@@ -187,15 +189,24 @@ const PriceSlider = ( {
 	 */
 	const priceInputOnBlur = useCallback(
 		( event ) => {
+			// Only refresh when finished editing the min and max fields.
+			if (
+				event.relatedTarget &&
+				event.relatedTarget.classList &&
+				event.relatedTarget.classList.contains(
+					'wc-block-price-filter__amount'
+				)
+			) {
+				return;
+			}
 			const isMin = event.target.classList.contains(
 				'wc-block-price-filter__amount--min'
 			);
-			const targetValue = event.target.value.replace( /[^0-9.-]+/g, '' );
-			const currentValues = isMin
-				? [ targetValue, maxPrice ]
-				: [ minPrice, targetValue ];
 			const values = constrainRangeSliderValues(
-				currentValues,
+				[
+					minPriceInput * 10 ** decimalScale,
+					maxPriceInput * 10 ** decimalScale,
+				],
```

Comment:
You should be able to just pass the current values here right?  I think as much as possible we should do all calculations/comparisons on the subunit values and only do conversion to currency representation with formatting on display.